mediaengine.go

```diff
@@ -26,34 +26,40 @@ const (
 	mediaNameVideo = "video"
 )
 
-// MediaEngine defines the codecs supported by a PeerConnection
+// A MediaEngine defines the codecs supported by a PeerConnection.
+// MediaEngines may be set up once and reused, including concurrently,
+// as long as they are unmodified.
 type MediaEngine struct {
 	codecs []*RTPCodec
 }
 
-// RegisterCodec registers a codec to a media engine
+// RegisterCodec adds codec to m.
+// RegisterCodec is not safe for concurrent use.
 func (m *MediaEngine) RegisterCodec(codec *RTPCodec) uint8 {
-	// pion/webrtc#43
+	// TODO: dynamically generate a payload type in the range 96-127 if one wasn't provided.
+	// See https://github.com/pion/webrtc/issues/43
 	m.codecs = append(m.codecs, codec)
 	return codec.PayloadType
 }
 
-// RegisterDefaultCodecs is a helper that registers the default codecs supported by Pion WebRTC
+// RegisterDefaultCodecs registers the default codecs supported by Pion WebRTC.
+// RegisterDefaultCodecs is not safe for concurrent use.
 func (m *MediaEngine) RegisterDefaultCodecs() {
-	// Audio Codecs in order of preference
+	// Audio Codecs in descending order of preference
 	m.RegisterCodec(NewRTPOpusCodec(DefaultPayloadTypeOpus, 48000))
 	m.RegisterCodec(NewRTPPCMUCodec(DefaultPayloadTypePCMU, 8000))
 	m.RegisterCodec(NewRTPPCMACodec(DefaultPayloadTypePCMA, 8000))
 	m.RegisterCodec(NewRTPG722Codec(DefaultPayloadTypeG722, 8000))
 
-	// Video Codecs in order of preference
+	// Video Codecs in descending order of preference
 	m.RegisterCodec(NewRTPVP8Codec(DefaultPayloadTypeVP8, 90000))
 	m.RegisterCodec(NewRTPVP9Codec(DefaultPayloadTypeVP9, 90000))
 	m.RegisterCodec(NewRTPH264Codec(DefaultPayloadTypeH264, 90000))
 }
 
-// PopulateFromSDP finds all codecs in a session description and adds them to a MediaEngine, using dynamic
-// payload types and parameters from the sdp.
+// PopulateFromSDP finds all codecs in sd and adds them to m, using the dynamic
+// payload types and parameters from sd.
+// PopulateFromSDP is not safe for concurrent use.
```

Comment:
Not sure how to word this properly, but `PopulateFromSDP` is meant to be used when you are answering. The offerer controls the PayloadTypes for the call, and this allows an answerer to match the PayloadTypes properly of the offerer.

You should also probably only call `PopulateFromSDP` once on a MediaEngine, and then use it only for that single session.

Reply:
Thanks. I took a stab and wordsmithing. Tell me what you think. It's slightly verbose, but I think it's better to be a bit too verbose than a bit too terse when it comes to docs.